### PR TITLE
Complex patch for TextArea and Emoji

### PR DESCRIPTION
**Pull request by CODeRUS**
+add EmojiDialog recent tab
+add EmojiDialog tapAndHold behaviour
*fix TextArea Emoji insertion
*fix TextArea Line break
*fix TextArea "<&>" marks
*fix Conversaion bubble text copy
*fix New message notification text

### CODE REVIEW

**Comment by CODeRUS** [comment]
fixing issue https://github.com/tgalal/wazapp/issues/120

**Comment by CODeRUS** [comment]
just have one issue with copy/paste multiline text
i will rewrite it soon

File: src/client/waxmpp.py [hunk 1]
```diff
@@ -916,8 +916,7 @@ def onAvailable(self):
 			self.interfaceHandler.call("presence_sendAvailable")
 		
 	
-	def sendMessage(self,jid,msg_text):
-		msg_text = self.strip(msg_text);
+	def sendMessage(self,jid,msg_text,count):
```
Comment (tgalal) [inline src/client/waxmpp.py:919]: count ?

Comment (CODeRUS) [inline src/client/waxmpp.py:919]: yeash. i stucked at holy bug when string received from last signal with garbage at the end ant can't solve in other way.
i have some ideas how to rewrite some things before that, will try tomorrow

Comment (tgalal) [inline src/client/waxmpp.py:919]: but what introduced that garbage? There was no garbage before

Comment (CODeRUS) [inline src/client/waxmpp.py:919]: i don't know
python or system
i think strip() eats it, i removed it for tests
okay, i found more easy and better way, will make changes tomorrow

for observing garbage you can look my next comment on this code

File: src/client/contacts.py [hunk 2]
```diff
@@ -110,9 +110,11 @@ def updateContacts(self,data):
 
 				if is_valid:
 					contact = self.store.Contact.getOrCreateContactByJid(jid)
+					length = len(newSatus)
+					newSatus = newSatus.encode('utf-8')
 					contact.status = newSatus
 					contact.save()
-					self.contactsRefreshSuccess.emit(self.mode, contact.status);	
+					self.contactsRefreshSuccess.emit(self.mode, newSatus, length);#holy bug. when sending string to it recieving wigh garbage at the end
```
Comment (CODeRUS) [inline src/client/contacts.py:117]: @tgalal you can check this place.
make codeDebug for newStatus before emit and after receiving
do tests with some strings. long strings with many emoji cause garbage after emiting before receiving

File: src/client/waxmpp.py (hunk 1 above)
Comment (tgalal) [inline src/client/waxmpp.py:919]: Fixed by escaping unicode characters before transmitting through signal and unescape in slot. New multibyte Emoji caused this. It occurs if transmitted as str, however if the string in inside a list or dictionary it works okay. So for str cases, escape unicode first. I updated sendMessage, setting status, changing group subject and update contact status signals. I removed any count parameters you added.

**Pull request merged**